### PR TITLE
[docs] Add dev install shell & docs about contributions and pull requests

### PR DESCRIPTION
**Pull request by lshoo**
To facilitate developers to participate and contribute to the Rooch Network project, relevant documents have been written, including:

Documentation on how to install the development environment and dependencies,
Documentation on how to contribute,
Guidelines and requirements for Pull Requests.

#27

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+## Scope
+
+This Code of Conduct applies within all project spaces, and it also applies when an individual is representing the project or its community in public spaces.  Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a project may be further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at conduct@rooch.network. All complaints will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident.  Further details of specific enforcement policies may be posted separately.
```
Comment (jolestar) [inline CODE_OF_CONDUCT.md:37]: add a mail link to conduct@rooch.network

File: docs/coding-conventions.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Move Coding Conventions
+
+This section lays out some basic coding conventions for Move that the Move team has found helpful. These are only recommendations, and you should feel free to use other formatting guidelines and conventions if you have a preference for them.
+
+## Naming
+
+- **Module names**: should be lower snake case, e.g., `fixed_point32`, `vector`.
+- **Type names**: should be camel case if they are not a native type, e.g., `Coin`, `RoleId`.
+- **Function names**: should be lower snake case, e.g., `destroy_empty`.
+- **Constant names**: should be upper camel case and begin with an `E` if they represent error codes (e.g., `EIndexOutOfBounds`) and upper snake case if they represent a non-error value (e.g., `MIN_STAKE`).
```
Comment (jolestar) [inline docs/coding-conventions.md:10]: #29  @geometryolife @pause125

File: scripts/dev-setup.sh [hunk 3]
```diff
@@ -0,0 +1,989 @@
...
+	  curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain stable
+    if [[ -n "${CARGO_HOME}" ]]; then
+      PATH="${CARGO_HOME}/bin:${PATH}"
+    else
+      PATH="${HOME}/.cargo/bin:${PATH}"
+    fi
+  fi
+}
+
+function install_vault {
```
[the remark this answers is not in the document]
Comment (lshoo) [reply scripts/dev-setup.sh]: Yes， removed them yet

**Review by jolestar — APPROVED** [review]

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (lshoo) [reply CODE_OF_CONDUCT.md]: Done

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+The Rooch Network project has adopted a Code of Conduct that we expect project participants to adhere to. Please read the following so that you can understand what actions will and will not be tolerated.
+
+## Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.
+
+## Our Standards
+
+Examples of behavior that contributes to creating a positive environment include:
```
Comment (pause125) [inline CODE_OF_CONDUCT.md:9]: to creating -> to create

File: docs/coding-conventions.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
+# Move Coding Conventions
+
+This section lays out some basic coding conventions for Move that the Move team has found helpful. These are only recommendations, and you should feel free to use other formatting guidelines and conventions if you have a preference for them.
```
Comment (pause125) [inline docs/coding-conventions.md:3]: I think we should unify coding conventions in this repo.

File: scripts/dev-setup.sh [hunk 6]
```diff
@@ -0,0 +1,989 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (lshoo) [reply scripts/dev-setup.sh]: Done

File: CODE_OF_CONDUCT.md (hunk 4 above)
Comment (lshoo) [reply CODE_OF_CONDUCT.md]: Done

File: docs/coding-conventions.md (hunk 5 above)
Comment (lshoo) [reply docs/coding-conventions.md]: Yes, add this

**Pull request merged**